xbmc/input/InputManager.cpp

```diff
@@ -491,7 +492,12 @@ bool CInputManager::OnEvent(XBMC_Event& newEvent)
         || (actionId >= ACTION_MOUSE_START && actionId <= ACTION_MOUSE_END))
         CApplicationMessenger::Get().SendAction(CAction(actionId, 0, newEvent.touch.x, newEvent.touch.y, newEvent.touch.x2, newEvent.touch.y2), WINDOW_INVALID, false);
     else
-      CApplicationMessenger::Get().SendAction(CAction(actionId), WINDOW_INVALID, false);
+    {
+      if (actionId == ACTION_BUILT_IN_FUNCTION && actionString.length() > 0)
```

Comment:
Is this distinction really necessary? Looking at the constructors you should be able to pass an empty action string / name and get the same result as with the other constructor right?